float32[] curr_state
float32[] tgt_state
---
float32[] action